Add support for the WISE1530 and integration framework for WICED SDK

geky opened the pull request:
Add support for the WISE1530 and integration framework for WICED SDK

This is a squash of a long series of work that introduces the WICED SDK into Mbed OS. Allowing the use of the WISE1530Interface in Mbed OS.

The WICED SDK requires additional steps to integrate into Mbed OS before it can be used. For more information see WICED.md.

cc @pan-, @SeppoTakalo, @SenRamakri, @sg-, @AnotherButler

0xc0170 commented on targets/TARGET_WICED/wiced_port/WICED/network/mbed/WWD/wwd_buffer.c:
```diff
@@ -0,0 +1,98 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#if defined (IAR_TOOLCHAIN)
```
should use IAR identifier rather here

why a header file inclusion is based on the toolchain?

geky replied:
This file had a lot of revisions, I'll see if I can remove it

0xc0170 commented on targets/TARGET_WICED/wiced_port/WICED/network/mbed/WWD/wiced_network.h:
```diff
@@ -0,0 +1,145 @@
+ * limitations under the License.
+ */
+#pragma once
+#if 0
```
this is external file and imported as is (so this deadcode is in there) ? same for typedef below

geky replied:
Oh good catch, I'll remove

0xc0170 commented on targets/TARGET_WICED/wiced_port/WICED/RTOS/mbed/WWD/wwd_rtos.cpp:
```diff
@@ -0,0 +1,304 @@
+#include "wwd_assert.h"
+#include "wwd_rtos_interface.h"
+#include "wiced_utilities.h"
+#include "mbed.h"
```
should not include this , pull in what is needed rather

geky replied:
Above comment got hidden by changes, but question still stands:

Is this a problem in cpp files? There's no risk of namespace polution since it's self contained in the cpp file. If anything this would help protect against filename changes.

SeppoTakalo replied:
Including only header files that you need will greatly reduce the rebuild time.

On the first build, dependency files are created. Then on the following builds, if there has not been any changes to those headers that you include, your file is not needed to rebuild.

But if everyone includes "mbed.h" this leads every small change to lead to full rebuild of the whole Mbed OS tree.

geky replied:
Fair enough, will update

0xc0170 commented on targets/TARGET_WICED/wiced_port/WICED/RTOS/mbed/WWD/wwd_rtos.cpp:
```diff
@@ -0,0 +1,304 @@
+#include "ticker_api.h"
+
+
+/**
```
are these docs needed here? Not done in the header file? If they are, this is duplicate and pollutes code file

geky replied:
They're copied from the header file, but the header file is hard to track down. I can remove them if we'd like.

0xc0170 commented on targets/TARGET_WICED/TARGET_WISE1530_F412RE/WISE1530Interface.cpp:
```diff
@@ -0,0 +1,340 @@
+
+#include "WISE1530Interface.h"
+
+#include "mbed.h"
```
dont include this header file outside of user space

geky replied:
Oh, is this a problem in cpp files? There's no risk of namespace polution since it's self contained in the cpp file. If anything this would help protect against filename changes.

cmonr replied:
As I understand it, a `name.cpp` file should only ever include its `name.h` file, for stylistic reasons, not technical.

geky commented on targets/TARGET_WICED/TARGET_WISE1530_F412RE/wise1530_emac.c:
```diff
@@ -0,0 +1,208 @@
+#include "mbed_assert.h"
```
Is updated

adbridge commented:
@0xc0170 Could you re-review please?

0xc0170 commented:
Please review travis, docs and tools report related errors.

geky commented:
I may need to help, @theotherjimmy:
```
Bad Assembler file name found: ./targets/TARGET_STM/TARGET_STM32F4/TARGET_Advantech_F412RE/device/TOOLCHAIN_GCC_ARM/startup_stm32f412rx.s
Bad Assembler file name found: ./targets/TARGET_STM/TARGET_STM32F4/TARGET_Advantech_F412RE/device/TOOLCHAIN_ARM_MICRO/startup_stm32f412rx.s
Bad Assembler file name found: ./targets/TARGET_STM/TARGET_STM32F4/TARGET_Advantech_F412RE/device/TOOLCHAIN_ARM_STD/startup_stm32f412rx.s
Bad Assembler file name found: ./targets/TARGET_STM/TARGET_STM32F4/TARGET_Advantech_F412RE/device/TOOLCHAIN_IAR/startup_stm32f412rx.s
```

What are valid assembly names?

```
AssertionError: Target WISE1530_F412RE contains invalid device_name Advantech_F412RE
E           assert 'Advantech_F412RE' in {'AC30M1332': {'algorithm': {'AC30M1x64/Flashloader/AC30M1x64_64.FLM': {'default': '1', 'ramsize': None, 'ramstart': N...header': 'AC30M1x64/Core/include/AC30M1x64.h'}, 'core': 'Cortex-M0', 'debug': 'AC30M1x64/SVD/AC30M1x64.svd', ...}, ...}
E            +  where 'Advantech_F412RE' = Target(name='WISE1530_F412RE', json_data={'WISE1530_F412RE': OrderedDict([('in...et', 2)], resolution_order_names=['WISE1530_F412RE', 'FAMILY_STM32', 'Target']).device_name
E            +  and   {'AC30M1332': {'algorithm': {'AC30M1x64/Flashloader/AC30M1x64_64.FLM': {'default': '1', 'ramsize': None, 'ramstart': N...header': 'AC30M1x64/Core/include/AC30M1x64.h'}, 'core': 'Cortex-M0', 'debug': 'AC30M1x64/SVD/AC30M1x64.svd', ...}, ...} = <tools.arm_pack_manager.Cache instance at 0x7fc444886f38>.index
```
This means I don't have a pack for the target I think? I'm not entirely sure what this response to this assertion should be.

geky commented:
Ah, sorry @AnotherButler I should have clarified. The WICED.md file is the only document worth editing:
https://github.com/geky/mbed/blob/3d44874f24460c23782d194b259d6efdfc776da7/targets/TARGET_WICED/WICED.md

The code you're editing predates me, I don't know if it's valuable to edit and I can't vouch that the edits are keeping the documentation's meaning correct sorry.

AnotherButler commented:
@geky Thanks for the clarification.

0xc0170 commented:
1. `.s` - suffix should be upper case to be portable, thus please rename them to have `.S`

2. Advantech_F412RE - does it have cmsis-pack? If yes, then is this a correct name?

geky commented:
Thanks, will update the assembly file.

I'll have to look into the cmsis pack thing more. Where does one normally find cmsis packs? Do they need to be generated?

0xc0170 commented:
> I'll have to look into the cmsis pack thing more. Where does one normally find cmsis packs? Do they need to be generated?

Yes, a vendor generates a pack and publishes it (device support as it is called there). You can use the arm pack manager that is part of this repo, or check in the uVision for instance (it has the pack installer to check if target is there). If there is not, we shall remove device_name. @theotherjimmy can provide more info in this area

KariHaapalehto commented:
Should we rename folder mbed-os\targets\TARGET_WICED\TARGET_WISE1530_F412RE\ and files under that folder? Those files are not hw specifig files just for wise1530, they are more like wiced adaptation to mbed and can be used with other variants too.

KariHaapalehto commented on targets/TARGET_WICED/WICED.md:
```diff
@@ -0,0 +1,229 @@
+- mbed_lib.json
+
+## Download the WICED SDK
+
```
Here are two different instructions, so I would add comment:
"Both WICED SDK and patch from Advantech is needed"

KariHaapalehto commented on targets/TARGET_WICED/WICED.md:
```diff
@@ -0,0 +1,229 @@
+on Advantech's Wiki:
+http://ess-wiki.advantech.com.tw/view/WISE-1530_SDK
+
+To get the WICED SDK, download WICED-Studio-5.2.0 for your platform:
```
Or WICED SDK WICED-Studio-5.2.0 for your platform can be get here:

KariHaapalehto commented on targets/TARGET_WICED/WICED.md:
```diff
@@ -0,0 +1,229 @@
+- mbed_lib.json
+
+## Download the WICED SDK
+
```
Here are two different instructions, so I would add comment:
"Both WICED SDK and patch from Advantech is needed"

geky replied:
Added to section header

KariHaapalehto commented on targets/TARGET_WICED/WICED.md:
```diff
@@ -0,0 +1,229 @@
+on Advantech's Wiki:
+http://ess-wiki.advantech.com.tw/view/WISE-1530_SDK
+
+To get the WICED SDK, download WICED-Studio-5.2.0 for your platform:
```
Or WICED SDK WICED-Studio-5.2.0 for your platform can be get here:

geky replied:
Hi @KariHaapalehto, I'm not exactly sure what the suggestion is for this line. Does the studio need to be named "WICED SDK WICED-Studio-5.2.0"?

geky replied:
Well, that's not right, the studio is WICED-Studio-5.2.0. Inside the studio you get the WICED SDK.

KariHaapalehto commented on targets/TARGET_WICED/WICED.md:
```diff
@@ -0,0 +1,229 @@
+To get the WICED SDK, download WICED-Studio-5.2.0 for your platform:
+https://community.cypress.com/community/wiced-wifi/wiced-wifi-documentation
+
+Once downloaded, move the WICED SDK folder into `targets/TARGET_WICED`.
```
Once downloaded, move the SDK's targets/TARGET_WICED/43xxx_Wi-Fi-folder into `targets/TARGET_WICED`.

geky replied:
Also added note about rename to WICED

KariHaapalehto commented on targets/TARGET_WICED/WICED.md:
```diff
@@ -0,0 +1,229 @@
+For the WISE1530, download the patch from Advantech:
+`WM-BN-BM-22_SDK_5.1.x_platform_patch.zip`
+
+Once downloaded, copy the contents into `targets/TARGET_WICED/WICED`.
```
Once downloaded, copy the contents into `targets/TARGET_WICED`.

geky replied:
It really should be `targets/TARGET_WICED/WICED` (after rename). Otherwise the libraries/platforms folders don't line up with their destinations. Note there are multiple nested WICED folders.

KariHaapalehto commented on targets/TARGET_WICED/WICED.md:
```diff
@@ -0,0 +1,229 @@
+
+1. Remove these files because they are unnecessary and large:
+   - `targets/TARGET_WICED/WICED/tools`
+   - `targets/TARGET_WICED/WICED/build`
```
There isn't build-directory

geky replied:
The build directory is generated if you compile using the WICED studio. I can go ahead and remove this if we want.

KariHaapalehto commented on targets/TARGET_WICED/WICED.md:
```diff
@@ -0,0 +1,229 @@
+   - `targets/TARGET_WICED/WICED/tools`
+   - `targets/TARGET_WICED/WICED/build`
+   - `targets/TARGET_WICED/WICED/apps`
+   - `targets/TARGET_WICED/WICED/factory`
```
There isn't factory-directory

geky replied:
Can remove

KariHaapalehto commented on targets/TARGET_WICED/WICED.md:
```diff
@@ -0,0 +1,229 @@
+   - `targets/TARGET_WICED/WICED/network/NoNS`
+   - `targets/TARGET_WICED/WICED/network/NuttX_NS`
+
+1. Add the `w_` prefix to all c/h files in WICED.
```
2.

geky replied:
Markdown automatically numbers this formatting. See:
https://github.com/geky/mbed/blob/fe39a2bf5064cb2b3d2a8f3e39b74beef23b2d02/targets/TARGET_WICED/WICED.md

KariHaapalehto commented on targets/TARGET_WICED/WICED.md:
```diff
@@ -0,0 +1,229 @@
+
+   This is required to avoid naming conflicts between Mbed OS and the WICED SDK.
+
+1. Prevent Mbed OS from compiling the include-only c files. These are
```
3.

KariHaapalehto commented on targets/TARGET_WICED/WICED.md:
````diff
@@ -0,0 +1,229 @@
+   - #include "wicedfs.h"
+   ```
+
+1. Create a C wrapper for the firmware image.
````
5.

KariHaapalehto commented on targets/TARGET_WICED/WICED.md:
```diff
@@ -0,0 +1,229 @@
+   Edit all c/h files, and change the relevant `#include` statements to use the
+   updated file names.
+
+1. Remove any references to the WICED file system.
```
4.

KariHaapalehto commented on targets/TARGET_WICED/WICED.md:
````diff
@@ -0,0 +1,229 @@
+   #endif
+   ```
+
+1. Create the `w_generated_security_dct.h` file that
````
6.

KariHaapalehto commented on targets/TARGET_WICED/WICED.md:
````diff
@@ -0,0 +1,229 @@
+   #define PRIVATE_KEY_STRING "\0"
+   ```
+
+1. Create the `w_generated_mac_address.txt` file that
````
7.

KariHaapalehto commented on targets/TARGET_WICED/WICED.md:
````diff
@@ -0,0 +1,229 @@
+   + WWD_RTOS_DEFINE_ISR( NoOS_systick_IRQHandler )
+   ```
+
+1. Comment out the bad `wwd_bus_packet_available_to_read` call while polling for
````
9.

KariHaapalehto commented on targets/TARGET_WICED/WICED.md:
```diff
@@ -0,0 +1,229 @@
+1. Comment out the bad `wwd_bus_packet_available_to_read` call while polling for
+   bus packet status.
+
+   In `targets/TARGET_WICED/WICED/WWD/internal/wwd_thread.c` line 336:
```
line 344

geky replied:
@SeppoTakalo pointed out we may be on different versions. It should have been marked as 4.1.*. With 4.1 are these lines still different?

KariHaapalehto commented on targets/TARGET_WICED/WICED.md:
```diff
@@ -0,0 +1,229 @@
+   Note this is required for the WISE1530 but may not be needed for other
+   platforms.
+
+1. Remove the includes of `w_core_cm4.h` and `w_core_cmFunc.h`, and replace the
```
10.

KariHaapalehto commented on targets/TARGET_WICED/WICED.md:
````diff
@@ -0,0 +1,229 @@
+   + #include "core_cm4.h"
+   ```
+
+1. Remove the padding of SDPCM data packets. This requires modifications in two
````
11.

KariHaapalehto commented on targets/TARGET_WICED/WICED.md:
````diff
@@ -0,0 +1,229 @@
+    } sdpcm_data_header_t;
+   ```
+
+   In `targets/TARGET_WICED/WICED/WWD/internal/w_wwd_sdpcm.c` line 1282:
````
line 1441:

KariHaapalehto commented on targets/TARGET_WICED/WICED.md:
```diff
@@ -0,0 +1,229 @@
+   Note this is required for the WISE1530 but may not be needed for other
+   platforms.
+
+1. Set up SDIO_ENUMERATION_TIMEOUT_MS to be configurable. Mbed OS will override this,
```
12

KariHaapalehto commented on targets/TARGET_WICED/WICED.md:
````diff
@@ -0,0 +1,229 @@
+     #define SDIO_ENUMERATION_TIMEOUT_MS          (500)
+   + #endif
+   ```
+
````
13. 12 Rename ./mbed-os/targets/TARGET_WICED/TARGET_WISE1530_F412RE/wise1530_emac.c to ./mbed-os/targets/TARGET_WICED/TARGET_WISE1530_F412RE/wise1530_emac.cpp

geky replied:
Why is this needed? wise1530_emac.c is in Mbed OS, so we can change it here if need be.

But why does it need to be changed to cpp, was there a compilation issue?

SeppoTakalo commented:
As this is only going for the feature branch and is a work in progress, I'm OK if this gets merged in as it is currently.

We will anyway continue on working to improve the instructions but at least we won't be blocking others to study it.

0xc0170 commented on features/FEATURE_BLE/targets/TARGET_WISE1530_F412RE/HCIDriver.cpp:
```diff
@@ -0,0 +1,428 @@
+#include <stdio.h>
+#include "CordioBLE.h"
+#include "CordioHCIDriver.h"
+#include "mbed.h"
```
what is needed here from mbed.h ? DigitalInout and PinNames ?

 missing license header file

geky replied:
looks like it was not used

cmonr commented:
@0xc0170 @SeppoTakalo @KariHaapalehto Are you all happy with @geky's updates?

SeppoTakalo commented:
I'm still in a process of going though the steps... I have not yet successfully build it.

SeppoTakalo commented:
Final problem that I cannot get past
```
./mbed-os/targets/TARGET_WICED/WICED/libraries/drivers/gspi/gspi_slave/w_gspi_slave.c:38:28: fatal error: platform_cache.h: No such file or directory
 #include "platform_cache.h"
```

This `platform_cache.h` file was one that was excluded from the build because of `.mbedignore` lists its folder `WICED/WICED/platform/ARM_CR4/`

SeppoTakalo commented:
Also, lots of files seems to be relying that `#pragma once` which is not working on GCC. Causing lots of warnings and some errors.

geky commented:
`#pragma once` should be supported in GCC, what version are you using?

We could remove the pragm once in `wiced_network.h` but not easily for the ones in the WICED library. Should we add a step to replace `#pragma once` instances with `#ifdef` gaurds?

geky commented:
Interesting, I don't have gspi in my version of WICED. May be a difference between 4.1.1 and 4.1.3 which is recent. Easy solution is to add gspi to .mbedignore

SeppoTakalo commented:
Finally:
```
Link: mbed-os-example-wifi
Elf2Bin: mbed-os-example-wifi
+------------------------+--------+-------+-------+
| Module                 |  .text | .data |  .bss |
+------------------------+--------+-------+-------+
| [fill]                 |    319 |     7 |    97 |
| [lib]/c.a              |  47369 |  2472 |    89 |
| [lib]/gcc.a            |   3368 |     0 |     0 |
| [lib]/misc             |    252 |    16 |    28 |
| [lib]/nosys.a          |     32 |     0 |     0 |
| esp8266-driver/ESP8266 |     16 |     0 |     0 |
| main.o                 |   1452 |     4 |    72 |
| mbed-os/drivers        |    910 |     4 |   100 |
| mbed-os/events         |   1576 |     0 |   101 |
| mbed-os/features       |  43949 |   173 | 34409 |
| mbed-os/hal            |   1581 |     4 |    66 |
| mbed-os/platform       |   2410 |     4 |   356 |
| mbed-os/rtos           |  13085 |   168 |  5989 |
| mbed-os/targets        | 413351 |    88 |  5085 |
| Subtotals              | 529670 |  2940 | 46392 |
+------------------------+--------+-------+-------+
Total Static RAM memory (data + bss): 49332 bytes
Total Flash memory (text + data): 532610 bytes

Object file w_http.o is not unique! It could be made from: ./mbed-os/targets/TARGET_WICED/WICED/libraries/protocols/HTTP/w_http.c ./mbed-os/targets/TARGET_WICED/WICED/libraries/protocols/HTTP_client/w_http.c
Image: ./BUILD/WISE1530_F412RE/GCC_ARM/mbed-os-example-wifi.bin
```

Couple of issues still remaining, I was forced to patch my local repository:

### Missing ignore directory
```diff
diff --git a/targets/TARGET_WICED/.mbedignore b/targets/TARGET_WICED/.mbedignore
index 5fae0d53c..95eb80950 100644
--- a/targets/TARGET_WICED/.mbedignore
+++ b/targets/TARGET_WICED/.mbedignore
@@ -107,6 +107,7 @@ WICED/platforms/BCM943907WCD2/
 WICED/platforms/BCM943362WCD4/
 WICED/platforms/BCM943362WCD8/
 WICED/platforms/BCM943907WCD1/
+WICED/platforms/BCMUSI22/
 WICED/libraries/daemons/device_configuration/w_device_configuration_http_content.c
 WICED/WICED/platform/MCU/STM32F4xx/WWD/wwd_SPI.c
 WICED/WICED/platform/MCU/STM32F4xx/w_platform_bootloader.c
```

### Mbed OS files that assume filenames before the rename

This, I'm not sure whether your instructions meant to patch Mbed OS also. I'm assuming that we were supposed to touch the Wiced SDK only, therefore these files need to be refactored to match the filenames your instructions would produce.

```diff
diff --git a/targets/TARGET_WICED/TARGET_WISE1530_F412RE/WISE1530Interface.cpp b/targets/TARGET_WICED/TARGET_WISE1530_F412RE/WISE1530Interface.cpp
index 6f5f7111a..f080b0012 100644
--- a/targets/TARGET_WICED/TARGET_WISE1530_F412RE/WISE1530Interface.cpp
+++ b/targets/TARGET_WICED/TARGET_WISE1530_F412RE/WISE1530Interface.cpp
@@ -19,10 +19,10 @@
 #include "nsapi.h"
 #include "lwipopts.h"
 
-#include "wiced_wifi.h"
-#include "wiced_management.h"
-#include "wwd_network_constants.h"
-#include "wwd_buffer_interface.h"
+#include "w_wiced_wifi.h"
+#include "w_wiced_management.h"
+#include "w_wwd_network_constants.h"
+#include "w_wwd_buffer_interface.h"
 
 #include "lwip/etharp.h"
 #include "lwip/ethip6.h"
diff --git a/targets/TARGET_WICED/TARGET_WISE1530_F412RE/wise1530_emac.c b/targets/TARGET_WICED/TARGET_WISE1530_F412RE/wise1530_emac.c
index e2ef61671..4b2703c57 100644
--- a/targets/TARGET_WICED/TARGET_WISE1530_F412RE/wise1530_emac.c
+++ b/targets/TARGET_WICED/TARGET_WISE1530_F412RE/wise1530_emac.c
@@ -19,10 +19,10 @@
 #include "lwipopts.h"
 #include <string.h>
 
-#include "wiced_wifi.h"
-#include "wiced_management.h"
-#include "wwd_network_constants.h"
-#include "wwd_buffer_interface.h"
+#include "w_wiced_wifi.h"
+#include "w_wiced_management.h"
+#include "w_wwd_network_constants.h"
+#include "w_wwd_buffer_interface.h"
 
 #include "lwip/etharp.h"
 #include "lwip/ethip6.h"
diff --git a/targets/TARGET_WICED/wiced_port/WICED/RTOS/mbed/WICED/w_rtos.h b/targets/TARGET_WICED/wiced_port/WICED/RTOS/mbed/WICED/w_rtos.h
index 4e53317b4..5b7b310a1 100644
--- a/targets/TARGET_WICED/wiced_port/WICED/RTOS/mbed/WICED/w_rtos.h
+++ b/targets/TARGET_WICED/wiced_port/WICED/RTOS/mbed/WICED/w_rtos.h
@@ -16,8 +16,8 @@
 #ifndef W_RTOS_H
 #define W_RTOS_H
 
-#include "wiced_result.h"
-#include "wiced_utilities.h"
+#include "w_wiced_result.h"
+#include "w_wiced_utilities.h"
 #include "wwd_rtos.h"
 
 #ifdef __cplusplus
diff --git a/targets/TARGET_WICED/wiced_port/WICED/RTOS/mbed/WICED/wiced_rtos.cpp b/targets/TARGET_WICED/wiced_port/WICED/RTOS/mbed/WICED/wiced_rtos.cpp
index 7df1819eb..a286b6650 100644
--- a/targets/TARGET_WICED/wiced_port/WICED/RTOS/mbed/WICED/wiced_rtos.cpp
+++ b/targets/TARGET_WICED/wiced_port/WICED/RTOS/mbed/WICED/wiced_rtos.cpp
@@ -13,10 +13,10 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-#include "wiced_result.h"
-#include "wiced_utilities.h"
-#include "wiced_rtos.h"
-#include "wiced_time.h"
+#include "w_wiced_result.h"
+#include "w_wiced_utilities.h"
+#include "w_wiced_rtos.h"
+#include "w_wiced_time.h"
 #include "wwd_rtos.h"
 #include "mbed.h"
 
diff --git a/targets/TARGET_WICED/wiced_port/WICED/RTOS/mbed/WWD/wwd_rtos.cpp b/targets/TARGET_WICED/wiced_port/WICED/RTOS/mbed/WWD/wwd_rtos.cpp
index 697041a31..015409235 100644
--- a/targets/TARGET_WICED/wiced_port/WICED/RTOS/mbed/WWD/wwd_rtos.cpp
+++ b/targets/TARGET_WICED/wiced_port/WICED/RTOS/mbed/WWD/wwd_rtos.cpp
@@ -14,10 +14,10 @@
  * limitations under the License.
  */
 #include "wwd_rtos.h"
-#include "wwd_constants.h"
-#include "wwd_assert.h"
-#include "wwd_rtos_interface.h"
-#include "wiced_utilities.h"
+#include "w_wwd_constants.h"
+#include "w_wwd_assert.h"
+#include "w_wwd_rtos_interface.h"
+#include "w_wiced_utilities.h"
 #include "rtos.h"
 #include "ticker_api.h"
 
@@ -36,7 +36,7 @@ extern "C" wwd_result_t host_rtos_create_thread(
 extern "C" wwd_result_t host_rtos_create_thread_with_arg(
         host_thread_type_t *p,
         void (*entry_function)(uint32_t),
-        const char *name, 
+        const char *name,
         void *stack,
         uint32_t stack_size,
         uint32_t priority,
diff --git a/targets/TARGET_WICED/wiced_port/WICED/network/mbed/WWD/wiced_network.h b/targets/TARGET_WICED/wiced_port/WICED/network/mbed/WWD/wiced_network.h
index 6d36a6095..a55c611d5 100644
--- a/targets/TARGET_WICED/wiced_port/WICED/network/mbed/WWD/wiced_network.h
+++ b/targets/TARGET_WICED/wiced_port/WICED/network/mbed/WWD/wiced_network.h
@@ -16,7 +16,7 @@
 #pragma once
 #include "w_tls_types.h"
 #include "w_dtls_types.h"
-#include "wiced_result.h"
+#include "w_wiced_result.h"
 
 #ifdef __cplusplus
 extern "C"
diff --git a/targets/TARGET_WICED/wiced_port/WICED/network/mbed/WWD/wwd_buffer.c b/targets/TARGET_WICED/wiced_port/WICED/network/mbed/WWD/wwd_buffer.c
index 8fd453851..8f96a8a37 100644
--- a/targets/TARGET_WICED/wiced_port/WICED/network/mbed/WWD/wwd_buffer.c
+++ b/targets/TARGET_WICED/wiced_port/WICED/network/mbed/WWD/wwd_buffer.c
@@ -18,9 +18,9 @@
 #include <stdlib.h>
 #include <assert.h>
 #include "wwd_buffer.h"
-#include "wwd_constants.h"
-#include "wwd_assert.h"
-#include "network/wwd_buffer_interface.h"
+#include "w_wwd_constants.h"
+#include "w_wwd_assert.h"
+#include "network/w_wwd_buffer_interface.h"
 #include "mbed_critical.h"
 
 wwd_result_t host_buffer_check_leaked(void)
```

SeppoTakalo commented:
Program starts, but is not able to initiate Wifi:
```
WiFi example

Scan:
Creating DCT file "DCT.bin"
Error opening DCT file "DCT.bin"
Creating DCT file "DCT.bin"
Error opening DCT file "DCT.bin"
```

Did I miss something in the instructions?

geky commented:
Is this with power-cycling the device (with the jumper?)

SeppoTakalo commented:
Yes, after power cycling the device.

geky commented:
> Missing ignore directory

Updated, thanks!

> Mbed OS files that assume filenames before the rename

Ah, the rename step was meant to say: "Add the `w_` prefix to all c/h files in WICED that do not begin with `wiced_` or `wwd_`."

I've updated the instructions, though do you think it would be easier to just add the `w_` prefix to all files without distinction?

0xc0170 commented:
@geky What is currently outstanding for this feature PR?

cmonr commented:
/morph build

SeppoTakalo commented:
This PR should be merged in as it will otherwise block any further development regarding Wiced based platforms.

Build failure comes from this exact platform, which cannot be build, until the whole sequence of manual steps have been completed, which are described in `WICED.md`.

0xc0170 commented:
>  Build failure comes from this exact platform, which cannot be build, until the whole sequence of manual steps have been completed, which are described in WICED.md.

@geky Should the target not have version defined in targets.json file? Would that resolve the build failure (it would not be built to enable further development) as not supported at the moment

geky commented:
Hmm, WICED is only needed for the network interface. Do you think we should add a config option to enabled the network interface? It would be able to compile as is without it.

SeppoTakalo commented:
@0xc0170 Please clarify, what is needed to skip this platform from the build?

It is incomplete, and requires human work to bring it in, so therefore CI should not try to build it.

0xc0170 commented:
First thought was to not have release version, but that is just connected to the release. Then public set to false in the targets.json, that would lead to unsupported MCU at the moment. 

> Hmm, WICED is only needed for the network interface. Do you think we should add a config option to enabled the network interface? It would be able to compile as is without it.

If we can do  that, seems better. the target code should compile and being tested, thus if we can skip network interface for now , sounds good to me

theotherjimmy commented:
>Hmm, WICED is only needed for the network interface. Do you think we should add a config option to enabled the network interface? It would be able to compile as is without it.

Would this be: don't do `features_add` of `LWIP`? Either way, I think this is the way to go.

geky commented:
Talked with @theotherjimmy and he suggested removing the `WICED` label from targets.json and adding a step in WICED.md to add a `target.extra_labels_add` to the `target_overrides` in the mbed_app.json of the user's program.

The alternative is to add `ifdefs` at the top of every file in the WICED wrapper. This approach is used primarily in the mbed drivers.

Thoughts?

I'll push up a patch with the target.extra_labels_add to move forward. We can change if we want.

SeppoTakalo commented:
I would vote for the first suggestion.
Disable the target in the `targets.json` and add that step into the `WICED.md`.

Having lots of `#ifdefs` quickly becomes painful to maintain.

But whatever works, is fine. This is anyway work-in-progress.

I would like to see this PR merged in quite quickly so that we can continue to add changes on top of it into the feature branch.

geky commented:
Updated, please review

0xc0170 commented:
I cherry picked a fix for travis for master, the feature branch should get branch after integration of this. Starting testing

/morph build

0xc0170 commented:
@geky I can not locate 2 symbols in the codebase on your branch that linker complains about, please review

SeppoTakalo commented:
If `FEATURE_WICED` is disabled, the `FEATURE_BLE` should be disabled as well.

KariHaapalehto commented on targets/TARGET_WICED/WICED.md:
````diff
@@ -0,0 +1,244 @@
+
+``` json
+{
+    "target_overrides": {
````
target_overrides isn't working in this case. Should it be:
-        "extra_labels_add": ["STM", "STM32F4", "Advantech_F412RE", "CORDIO"],
+        "extra_labels_add": ["STM", "STM32F4", "Advantech_F412RE", "CORDIO", "WICED"],

SeppoTakalo replied:
`"target.extra_labels_add": ["WICED"]` works for me.

SeppoTakalo commented:
@0xc0170 Please restart the test.

0xc0170 commented:
/morph build

geky commented:
Looks like it's almost good to go. @SeppoTakalo thanks for fixing that 👍

geky commented:
@0xc0170, @cmonr, tests already failing for NUCLEO_F746ZG:
```
tests-mbed_drivers-lp_ticker
:170::FAIL: Values Not Within Delta 2000 Expected 10000 Was 12546
```

Is there something on master that makes lp-ticker more stable? or is this just a case of rerun?

cmonr commented:
I'll restart it. @geky, the only thing that I could really think of is if the CI machine running the test is being hit hard, and having issues keeping up with all of the communication with the Device Under Test.

/morph uvisor-test

cmonr commented:
/morph test

cmonr commented:
Sorry, but it looks like this PR is doing something to lp_ticker test with the NUCLEO_F746ZG. The device is working fine, as evidenced by its previous build (https://github.com/ARMmbed/mbed-os/pull/5022#issuecomment-362400419).

Results are about to be reported.

kjbracey-arm commented:
Please rebase and resubmit - hopefully the cause of the failures should have been cleared on master.

0xc0170 commented:
/morph build

geky commented:
@cmonr, @0xc0170, well, it passes the NUCLEO_F746ZG now.

Are there known issues with UBLOX_EVK_ODIN_W2 tests-mbed_hal-ticker? I'm not sure how this pr could have touched that device either.

geky commented:
🎉
